expression/integration_test.go

```diff
@@ -6386,6 +6386,16 @@ func (s *testIntegrationSuite) TestCollation(c *C) {
 	tk.MustQuery("select collation(@test_collate_var)").Check(testkit.Rows("utf8mb4_general_ci"))
 	tk.MustExec("set @test_collate_var = concat(\"a\", \"b\" collate utf8mb4_bin)")
 	tk.MustQuery("select collation(@test_collate_var)").Check(testkit.Rows("utf8mb4_bin"))
+
+	tk.MustQuery("select locate('1', '123' collate utf8mb4_bin, 2 collate `binary`);").Check(testkit.Rows("0"))
+	tk.MustQuery("select locate('1', '123' collate utf8mb4_bin, 2 collate `binary`);").Check(testkit.Rows("0"))
```

Comment:
It is duplicated with line6390.

Reply:
Do we need to add some error tests like 
```
mysql> select locate('1' collate utf8mb4_general_ci, '123' collate utf8mb4_bin, 1 collate `binary`);
ERROR 1267 (HY000): Illegal mix of collations (utf8mb4_bin,EXPLICIT) and (utf8mb4_general_ci,EXPLICIT) for operation 'locate'
```

Reply:
updated

Reply:
> Do we need to add some error tests like

> mysql> select locate('1' collate utf8mb4_general_ci, '123' collate utf8mb4_bin, 1 collate `binary`);

> ERROR 1267 (HY000): Illegal mix of collations (utf8mb4_bin,EXPLICIT) and (utf8mb4_general_ci,EXPLICIT) for operation 'locate'

I mean add this type of testing, not just `locate`.

Reply:
I think it is better to add the tests in the next PR because the next PR will make the aggregate the same with mysql.